pkg/bitcoin/address_test.go

```diff
@@ -192,328 +80,250 @@ func TestEncodeAddress(t *testing.T) {
+				0xFF, 0xFF, 0xFF, 0xFF, 0xFE, 0xFF, 0xFF, 0xFD,
+				0x2F, 0xb2, 0xe0, 0xea, 0xdd, 0xfb, 0x84, 0xcc,
+				0xf9, 0x74, 0x44, 0x64, 0xf8, 0x2e, 0x16, 0x0b,
+				0xfa, 0x9b, 0x8b, 0x64, 0xf9, 0xd4, 0xc0, 0x3f,
+				0x99, 0x9b, 0x86, 0x43, 0xf6, 0x56, 0xb4, 0x12,
+				0xa3,
+			},
+			encoding:    NativeSegwit,
+			chainParams: Mainnet,
+			wantErr:     errors.New("pubkey X parameter is >= to P"),
 		},
 		{
 			// https://github.com/decred/dcrd/blob/b60c60f/dcrec/secp256k1/pubkey_test.go#L105-L109
 			name: "invalid public key (not on curve)",
-			request: &pb.EncodeAddressRequest{
-				PublicKey: []byte{
-					0x03, 0xce, 0x0b, 0x14, 0xfb, 0x84, 0x2b, 0x1b,
-					0xa5, 0x49, 0xfd, 0xd6, 0x75, 0xc9, 0x80, 0x75,
-					0xf1, 0x2e, 0x9c, 0x51, 0x0f, 0x8e, 0xf5, 0x2b,
-					0xd0, 0x21, 0xa9, 0xa1, 0xf4, 0x80, 0x9d, 0x3b,
-					0x4c,
-				},
-				Encoding:    pb.AddressEncoding_ADDRESS_ENCODING_P2WPKH,
-				ChainParams: mainnetChainParamsProto,
-			},
-			want: nil,
-			wantErr: status.New(codes.InvalidArgument,
-				"invalid square root"), // FIXME: Improve error in btcd
+			publicKey: []byte{
+				0x03, 0xce, 0x0b, 0x14, 0xfb, 0x84, 0x2b, 0x1b,
+				0xa5, 0x49, 0xfd, 0xd6, 0x75, 0xc9, 0x80, 0x75,
+				0xf1, 0x2e, 0x9c, 0x51, 0x0f, 0x8e, 0xf5, 0x2b,
+				0xd0, 0x21, 0xa9, 0xa1, 0xf4, 0x80, 0x9d, 0x3b,
+				0x4c,
+			},
+			encoding:    NativeSegwit,
+			chainParams: Mainnet,
+			wantErr:     errors.New("invalid square root"), // FIXME: Improve error in btcd
+		},
+		{
+			name: "ErrUnknownAddressType",
+			publicKey: []byte{
+				0x02, 0x53, 0x0c, 0x54, 0x8d, 0x40, 0x26, 0x70,
+				0xb1, 0x3a, 0xd8, 0x88, 0x7f, 0xf9, 0x9c, 0x29,
+				0x4e, 0x67, 0xfc, 0x18, 0x09, 0x7d, 0x23, 0x6d,
+				0x57, 0x88, 0x0c, 0x69, 0x26, 0x1b, 0x42, 0xde,
+				0xf7,
+			},
+			encoding:    9999,
+			chainParams: Mainnet,
+			wantErr:     ErrUnknownAddressType,
 		},
 	}
 
-	ctx := context.Background()
 	s := &Service{}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := s.EncodeAddress(ctx, tt.request)
-			grpcErr := status.Convert(err)
-
-			if grpcErr != nil && tt.wantErr == nil {
-				t.Errorf("unexpected error in EncodeAddress(): %v", grpcErr.Message())
+			got, err := s.EncodeAddress(tt.publicKey, tt.encoding, tt.chainParams)
+			if err != nil && tt.wantErr == nil {
+				t.Errorf("unexpected error in EncodeAddress(): %v", err)
 				return
 			}
 
-			if tt.wantErr != nil {
-				if grpcErr.Code() != tt.wantErr.Code() {
-					t.Errorf("EncodeAddress() gRPC error code = %v, want %v",
-						grpcErr.Code(), tt.wantErr.Code())
-					return
-				}
+			if err == nil && tt.wantErr != nil {
+				t.Errorf("unexpected success in EncodeAddress(): want error %v",
+					tt.wantErr)
+				return
+			}
 
-				if grpcErr.Message() != tt.wantErr.Message() {
-					t.Errorf("EncodeAddress() gRPC error msg = %v, want %v",
-						grpcErr.Message(), tt.wantErr.Message())
-					return
-				}
+			if err != nil && tt.wantErr.Error() != err.Error() {
```

Comment:
I you only want to test error message, just store the string in `wantErr` and write:
```
if err != nil && tt.wantErr != err.Error() {
```

Reply:
See my response [here](https://github.com/LedgerHQ/lama-bitcoin-svc/pull/9/files#r475756875).